### PR TITLE
Threading Update (still needs work)

### PR DESCRIPTION
**Pull request by avikbmohan**

### CODE REVIEW

File: integration_test.py [hunk 1]
```diff
@@ -18,35 +19,55 @@ def pas():
...
+line()
 ## Kill Server ##
-subprocess.call('kill ' + pid, shell = True)
+print("\nKilling Server.\n")
+serverProcess.kill()
 
 
 ## Conclusion ##
 print("Passed all tests?\n" + str(passing) + '\n')
+#if passing:
```
Comment (richardmin) [inline integration_test.py]: Why does the integration test always succeed?

**Review by richardmin — CHANGES_REQUESTED** [review]

**Comment by richardmin** [comment]
The integration test still always fails on echo

**Comment by richardmin** [comment]
There's a bug with ServerStatus -- it's global and should use mutexes.

File: integration_test.py [hunk 2]
```diff
@@ -17,40 +20,66 @@ def nameTest(name):
...
-nameTest("Connection")
-command2 = 'curl -Is localhost:' + str(port) + " | cat"
-output=subprocess.check_output(command2, shell=True)
-if output != "":
-    pas()
-    print(output)
+    print('Sending echo request after blocking request.\n')
+    q = telnetlib.Telnet('localhost', 2028, 2)
+    q.write(req2.encode('ascii'))
+    print('Sending echo request after blocking request.\n')
```
Comment (avikbmohan) [inline integration_test.py]: I added this line for debugging, will remove

File: integration_test.py [hunk 3]
```diff
@@ -17,40 +20,66 @@ def nameTest(name):
...
+    result = q.read_all().decode('utf-8')
+    print("Result of echo request:\n")
+    print(result)
+    if result != req2:
+        passing = False
```
Comment (avikbmohan) [inline integration_test.py]: Also I know this condition is wrong, gotta change it

**Comment by richardmin** [comment]
You should probably delete integration_test2.py, unless that's useful.

**Review by richardmin — APPROVED** [review]

**Pull request merged**